crates/project-model/src/project_json.rs

```diff
@@ -54,7 +54,11 @@ use std::path::PathBuf;
 use base_db::{CrateDisplayName, CrateId, CrateName, Dependency, Edition};
 use paths::{AbsPath, AbsPathBuf};
 use rustc_hash::FxHashMap;
-use serde::{de, Deserialize};
+use serde::{
+    de,
+    Deserialize::{self},
```

Comment:
Typo?

Reply:
The `::{self}` is actually required (it explicitly only imports the type namespace), without it this will import both the derive attribute and the trait, while we already have the derive attribute in scope via the `serde_derive::Deserialize` import (which is only the case because the `derive` feature is still enabled thanks to the deps).

I'll move this PR to draft though and wait until we can maybe get our deps to move, then we don't need these path shenanigans.